src/cryptosettings.h

```diff
@@ -5,35 +5,48 @@
 #ifndef CRYPTOSETTINGS_H
 #define CRYPTOSETTINGS_H
 
+#include <QByteArray>
 #include <QSettings>
 
 constexpr int CRYPTO_SETTINGS_KEY_SIZE = 32;
 
-class CryptoSettings final {
+class CryptoSettings {
  public:
   enum Version {
     NoEncryption,
     EncryptionChachaPolyV1,
+    EncryptionChachaPolyV2,
   };
 
+  explicit CryptoSettings();
```

Comment:
this could be protected instead of public, given we expect a factory `format()` to create it,no ?

Reply:
Actually, I don't think it's even possible to instantiate the base class at all given that it has unimplemented virtual functions. I'll make it protected.

For the unit tests we actually just create the derived class directly, so there is one niche use case where we don't use `CryptoSettings::format()`